### PR TITLE
dns: add multi cluster and multi network cases

### PR DESCRIPTION
**Pull request by ramaraochavali**
Adds some multi cluster and multi network cases for DNS

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[X] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/name_table_test.go [hunk 1]
```diff
@@ -143,6 +174,7 @@ func TestNameTable(t *testing.T) {
 		name              string
 		proxy             *model.Proxy
 		push              *model.PushContext
+		mcheadless        bool
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/name_table_test.go:177]: enableMultiClusterHedless

File: pilot/pkg/networking/core/v1alpha3/name_table_test.go [hunk 2]
```diff
@@ -163,8 +195,132 @@ func TestNameTable(t *testing.T) {
...
+						Ips:       []string{"1.2.3.4", "19.6.7.8", "9.16.7.8"},
+						Registry:  "Kubernetes",
+						Shortname: "headless-svc",
+						Namespace: "testns",
+					},
+				},
+			},
+		},
+		{
+			name:  "headless service pods with cluster isolation",
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/name_table_test.go:253]: what is `cluster isolation`?

Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/name_table_test.go]: i meant multi cluster- changed

**Review by hzxuzhonghu — APPROVED** [review]

**Review by howardjohn — COMMENTED** [review]
lgtm

**Pull request merged**